File: client/src/types/horoscope.ts
export interface HoroscopeDetail {
  love: string
  career: string
  wealth: string
  health: string
}

export interface HoroscopeResult {
  zodiac: string
  zodiac_cn: string
  period: 'daily' | 'weekly' | 'monthly'
  date: string
  overall: number
  love: number
  career: number
  wealth: number
  health: number
  lucky_number: number
  lucky_color: string
  summary: string
  detail: HoroscopeDetail
}

export interface ZodiacSign {
  name: string
  name_cn: string
  symbol: string
  element: 'fire' | 'earth' | 'air' | 'water'
  dateRange: string
}

export const ZODIAC_SIGNS: ZodiacSign[] = [
  { name: 'aries', name_cn: '白羊座', symbol: '♈', element: 'fire', dateRange: '3.21 - 4.19' },
  { name: 'taurus', name_cn: '金牛座', symbol: '♉', element: 'earth', dateRange: '4.20 - 5.20' },
  { name: 'gemini', name_cn: '双子座', symbol: '♊', element: 'air', dateRange: '5.21 - 6.21' },
  { name: 'cancer', name_cn: '巨蟹座', symbol: '♋', element: 'water', dateRange: '6.22 - 7.22' },
  { name: 'leo', name_cn: '狮子座', symbol: '♌', element: 'fire', dateRange: '7.23 - 8.22' },
  { name: 'virgo', name_cn: '处女座', symbol: '♍', element: 'earth', dateRange: '8.23 - 9.22' },
  { name: 'libra', name_cn: '天秤座', symbol: '♎', element: 'air', dateRange: '9.23 - 10.23' },
  { name: 'scorpio', name_cn: '天蝎座', symbol: '♏', element: 'water', dateRange: '10.24 - 11.22' },
  { name: 'sagittarius', name_cn: '射手座', symbol: '♐', element: 'fire', dateRange: '11.23 - 12.21' },
  { name: 'capricorn', name_cn: '摩羯座', symbol: '♑', element: 'earth', dateRange: '12.22 - 1.19' },
  { name: 'aquarius', name_cn: '水瓶座', symbol: '♒', element: 'air', dateRange: '1.20 - 2.18' },
  { name: 'pisces', name_cn: '双鱼座', symbol: '♓', element: 'water', dateRange: '2.19 - 3.20' },
]

export const ELEMENT_COLORS: Record<string, string> = {
  fire: '#ef4444',
  earth: '#a3e635',
  air: '#38bdf8',
  water: '#818cf8',
}

File: server/internal/model/horoscope.go
package model

// ZodiacInfo holds static zodiac data.
type ZodiacInfo struct {
	Name    string `json:"name"`    // e.g. "aries"
	NameCN  string `json:"name_cn"` // e.g. "白羊座"
	Element string `json:"element"` // fire/earth/air/water
	Planet  string `json:"planet"`  // ruling planet
}

// Zodiacs is the ordered list of all 12 zodiac signs.
var Zodiacs = []ZodiacInfo{
	{Name: "aries", NameCN: "白羊座", Element: "fire", Planet: "火星"},
	{Name: "taurus", NameCN: "金牛座", Element: "earth", Planet: "金星"},
	{Name: "gemini", NameCN: "双子座", Element: "air", Planet: "水星"},
	{Name: "cancer", NameCN: "巨蟹座", Element: "water", Planet: "月亮"},
	{Name: "leo", NameCN: "狮子座", Element: "fire", Planet: "太阳"},
	{Name: "virgo", NameCN: "处女座", Element: "earth", Planet: "水星"},
	{Name: "libra", NameCN: "天秤座", Element: "air", Planet: "金星"},
	{Name: "scorpio", NameCN: "天蝎座", Element: "water", Planet: "冥王星"},
	{Name: "sagittarius", NameCN: "射手座", Element: "fire", Planet: "木星"},
	{Name: "capricorn", NameCN: "摩羯座", Element: "earth", Planet: "土星"},
	{Name: "aquarius", NameCN: "水瓶座", Element: "air", Planet: "天王星"},
	{Name: "pisces", NameCN: "双鱼座", Element: "water", Planet: "海王星"},
}

// HoroscopeResult is the complete horoscope response.
type HoroscopeResult struct {
	RecordID    uint            `json:"record_id,omitempty"`
	Zodiac      string          `json:"zodiac"`
	ZodiacCN    string          `json:"zodiac_cn"`
	Period      string          `json:"period"`
	Date        string          `json:"date"`
	Overall     int             `json:"overall"`
	Love        int             `json:"love"`
	Career      int             `json:"career"`
	Wealth      int             `json:"wealth"`
	Health      int             `json:"health"`
	LuckyNumber int             `json:"lucky_number"`
	LuckyColor  string          `json:"lucky_color"`
	Summary     string          `json:"summary"`
	Detail      HoroscopeDetail `json:"detail"`
}

// HoroscopeDetail holds dimension-specific text.
type HoroscopeDetail struct {
	Love   string `json:"love"`
	Career string `json:"career"`
	Wealth string `json:"wealth"`
	Health string `json:"health"`
}

// ZodiacLookup looks up a zodiac by English name (case-insensitive).
// Returns nil if not found.
func ZodiacLookup(name string) *ZodiacInfo {
	for i := range Zodiacs {
		if Zodiacs[i].Name == name {
			return &Zodiacs[i]
		}
	}
	return nil
}
